Sources/UAPusher/Builder/Audience.swift

```diff
@@ -0,0 +1,63 @@
+import JSON
+
+/// Audience segment of an Urban Airship push notification payload
+/// See: https://docs.urbanairship.com/api/ua/#audience-selectors
+public struct Audience: Segment {
+    
+    // MARK: Class fields
+    
+    public let key: String = "audience"
+    public var payload: JSON
+    
+    // MARK: Selector values
+    
+    public enum Selector: JSONRepresentable {
+        case all
+        case tag(value: String)
+        case tags(values: [String])
+        case namedUser(value: String)
+        case alias(value: String)
+        
+        public func makeJSON() throws -> JSON {
+            switch self {
+            case .all:
+                return "all"
+            case .tag(let value):
+                var json: JSON = JSON()
```

Comment:
A minor thing, but consider moving this `JSON` init to above the switch to save a couple of lines (within each case)

Reply:
since I don't need it in the .all case I disagree here